Code/src/main/java/nl/utwente/group10/haskell/type/VarT.java

```diff
@@ -156,4 +156,9 @@ public final String toString() {
 
         return intersection;
     }
+
+    @Override
+    public int compareTo(final Type type) {
+        return this.equals(type) ? 0 : -1;
```

Comment:
Yes, that is the expected behaviour. I think the cleanest solution is to override ```equals``` and simply call ```super.equals``` and explain this in the javadoc.